src/client/utils/hooks/useQuoteCartData.ts

```diff
@@ -1,21 +1,212 @@
 import { useParams } from 'react-router-dom'
 import { useQuoteCartQuery } from 'data/graphql'
 import { useCurrentLocale } from 'l10n/useCurrentLocale'
+import { getMainQuote } from 'api/quoteBundleSelectors'
+import { getSelectedBundleVariant } from 'api/quoteCartQuerySelectors'
+import {
+  typeOfResidenceTextKeys,
+  HomeInsuranceTypeOfContract,
+} from 'pages/OfferNew/utils'
+import { GenericQuoteData } from '/pages/OfferNew/types'
+import { HouseQuoteDetails } from 'src/client/api/quoteDetailsDataTypes'
+import { isHouseQuoteDetailsData } from 'api/quoteDetailsDataSelectors'
+import { formatPostalNumber } from '../postalNumbers'
+import { useSelectedInsuranceTypes } from './useSelectedInsuranceTypes'
+
+export type Value = {
+  value?: string | number
+  textKey?: string
+  prefix?: string
+  suffix?: string
+}
+
+type Details = {
+  label: string
+  value: Value
+}
+
+type DetailsGroup = Details[]
 
 export const useQuoteCartData = () => {
   const { id: quoteCartId } = useParams<{ id: string }>()
   const { isoLocale } = useCurrentLocale()
-
   const { data, loading, error } = useQuoteCartQuery({
-    variables: {
-      id: quoteCartId,
-      locale: isoLocale,
-    },
+    variables: { id: quoteCartId, locale: isoLocale },
   })
 
+  const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
+  const selectedQuoteBundle = getSelectedBundleVariant(
+    data,
+    selectedInsuranceTypes,
+  )
+  if (!selectedQuoteBundle) return null
+
+  const prices = selectedQuoteBundle?.bundle.quotes.map((item) => {
+    return { displayName: item.displayName, price: item.price.amount }
+  })
+
+  const mainQuote = getMainQuote(selectedQuoteBundle.bundle)
+
+  const priceData = {
+    prices: prices,
+    discount: selectedQuoteBundle.bundle.bundleCost.monthlyDiscount.amount,
+    currency: selectedQuoteBundle.bundle.bundleCost.monthlyNet.currency,
+    totalBundleCost: selectedQuoteBundle.bundle.bundleCost.monthlyNet.amount,
+    campaignName: data?.quoteCart.campaign?.displayValue,
+  }
+  const quoteDetails = mainQuote.data
+
+  const getHomeDetails = (): DetailsGroup => {
```

Comment:
Redefining these inside the hook will not be good for performance (at least not without `useCallback`) - consider moving them outside the hook :)